webapp/channels/src/components/new_search/search_box_hints.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useSelector} from 'react-redux';
+
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+import type {ProviderResult} from 'components/suggestion/provider';
+import SearchDateSuggestion from 'components/suggestion/search_date_suggestion';
+
+import type {GlobalState} from 'types/store';
+
+import SearchHints from './search_hint';
+
+type Props = {
+    searchTerms: string;
+    setSearchTerms: (searchTerms: string) => void;
+    searchType: string;
+    selectedOption: number;
+    providerResults: ProviderResult<unknown>|null;
+    focus: (pos: number) => void;
+}
+
+const SearchBoxHints = ({searchTerms, setSearchTerms, searchType, providerResults, selectedOption, focus}: Props) => {
+    const license = useSelector(getLicense);
+    let SearchPluginHints = useSelector((state: GlobalState) => state.plugins.components.SearchHints) || [];
+    if (license.IsLicensed !== 'true') {
+        SearchPluginHints = [];
+    }
+
+    if (searchType === '' || searchType === 'messages' || searchType === 'files') {
+        return (
+            <SearchHints
+                onSelectFilter={(filter: string) => {
+                    if (searchTerms.endsWith(' ') || searchTerms.length === 0) {
+                        setSearchTerms(searchTerms + filter);
+                        focus(searchTerms.length + filter.length);
+                    } else {
+                        setSearchTerms(searchTerms + ' ' + filter);
+                        focus(searchTerms.length + filter.length + 1);
+                    }
+                }}
+                searchType={searchType}
+                searchTerms={searchTerms}
+                hasSelectedOption={Boolean(providerResults && providerResults.items.length > 0 && selectedOption !== -1)}
+                isDate={providerResults?.component === SearchDateSuggestion}
+            />
+        );
+    }
+
+    const pluginComponentInfo = SearchPluginHints.find(({pluginId}: any) => {
+        if (searchType === pluginId) {
+            return true;
+        }
+        return false;
+    });
+
+    if (!pluginComponentInfo) {
+        return null;
+    }
+
+    const Component: any = pluginComponentInfo.component;
+
+    return (
+        <Component
+            key={pluginComponentInfo.pluginId}
+            onChangeSearch={(value: string, matchedPretext: string) => {
+                const changedValue = value.replace(matchedPretext, '');
+                setSearchTerms(searchTerms + changedValue + ' ');
+                focus(searchTerms.length + changedValue.length + 1);
+            }}
+            searchTerms={searchTerms}
+        />
```

Comment:
I know we don't usually do this, but I wonder if we should start adding error boundaries to these "arbitrary component executions".

Reply:
very good call, adding them.